crates/containerd-shimkit/src/sandbox/instance_utils.rs

```diff
@@ -0,0 +1,112 @@
+//! Common utilities for the containerd shims.
+
+use std::fs::File;
+use std::io::{Error as IoError, ErrorKind, Result as IoResult};
+use std::path::PathBuf;
+
+use serde::{Deserialize, Serialize};
+
+use super::{Error, InstanceConfig};
+use crate::sys::DEFAULT_CONTAINER_ROOT_DIR;
+use crate::sys::stdio::open;
+
+#[derive(Serialize, Deserialize)]
+struct Options {
+    root: Option<PathBuf>,
+}
+
+impl InstanceConfig {
```

Comment:
Why doesn't this appear where `InstanceConfig` is defined?

Reply:
Originally these were free functions defined in this file, and not members of `InstanceConfig`.
I agree that it would be better where `InstanceConfig` is defined, but I would prefer doing that in a different PR.
I think there's a lot of cleanup we can do in this crate, including shuffling things around, and renaming stuff (like the `Cli` struct, the `Local` struct, etc).